packages/foundry-sdk-generator/src/generate/betaClient/copyFiles.ts

```diff
@@ -45,73 +46,33 @@ export async function copyFiles(
 
     const sourceFile = project.createSourceFile(newModulePath, file);
 
-    for (const importDeclaration of sourceFile.getImportDeclarations()) {
-      const moduleSpecifier = importDeclaration.getModuleSpecifier();
-      const newModuleSpecifier = transformModuleSpecifier(
-        moduleSpecifier.getLiteralValue(),
-        newModulePath,
-      );
-      moduleSpecifier.setLiteralValue(newModuleSpecifier);
-
-      if (newModuleSpecifier.startsWith("internal")) {
-        for (
-          const importName of importDeclaration.getNamedImports().map(imp =>
-            imp.getName()
-          )
-        ) {
-          importSet.add(importName);
-        }
-      }
-    }
-
-    for (const exportedDecl of sourceFile.getStatements()) {
-      if (Node.isModifierable(exportedDecl)) {
-        exportedDecl.toggleModifier("declare", false);
-      }
-    }
-
-    for (const exportDeclaration of sourceFile.getExportDeclarations()) {
-      const moduleSpecifier = exportDeclaration.getModuleSpecifier();
-      if (moduleSpecifier) {
-        const newModuleSpecifier = transformModuleSpecifier(
-          moduleSpecifier.getLiteralValue(),
-          newModulePath,
-        );
-        moduleSpecifier.setLiteralValue(newModuleSpecifier);
-        if (newModuleSpecifier.startsWith("internal")) {
-          for (
-            const exportName of exportDeclaration.getNamedExports().map(exp =>
-              exp.getName()
-            )
-          ) {
-            importSet.add(exportName);
-          }
-        }
-      }
-    }
+    remapImportsExports(sourceFile, newModulePath, importSet);
```

Comment:
This was extracted to the method below with the addition of handling dynamic imports.